src/scripts/main.js

```diff
@@ -1,3 +1,227 @@
 'use strict';
 
-// write code here
+function updateSortDirection(columnIndex) {
+  if (columnIndex === lastSortedColumn) {
+    sortDirection = -sortDirection;
+  } else {
+    sortDirection = 1;
+  }
+
+  lastSortedColumn = columnIndex;
+}
+
+function getCellValue(row, columnIndex) {
+  return currencyToNumber(row.cells[columnIndex].innerText);
+}
+
+function currencyToNumber(value) {
+  if (value[0] === '$') {
+    return +value.slice(1).split(',').join('');
+  }
+
+  return isNaN(value) ? value : +value;
+}
+
+function sortTable(columnIndex) {
+  const rows = [...tbody.querySelectorAll('tr')];
+
+  rows.sort((row1, row2) => {
+    const value1 = getCellValue(row1, columnIndex);
+    const value2 = getCellValue(row2, columnIndex);
+
+    return typeof value1 === 'number'
+      ? (value1 - value2) * sortDirection
+      : value1.toString().localeCompare(value2.toString()) * sortDirection;
```

Comment:
The sorting logic assumes that all values are either numbers or strings. If there are other types of data, this could lead to unexpected behavior. Consider adding additional checks or handling for other data types.